lib/repl.js

```diff
@@ -779,7 +747,12 @@ function REPLServer(prompt,
       self.lines.level = [];
       self.displayPrompt();
     }
-  });
+  };
+
+  // Set to null first to prevent ERR_UNCAUGHT_EXCEPTION_CAPTURE_ALREADY_SET from
+  // being thrown on set.
+  process.setUncaughtExceptionCaptureCallback(null);
+  process.setUncaughtExceptionCaptureCallback((er) => self._onEvalError(er));
```

Comment:
we should really not be setting this in an class we can instantiate multiple times

Reply:
I'm not really sure what's the best way to acheive this.

If it's set the global level, any uncaughtExceptions that don't have a stack will be sent to *all* active REPLs.

Reply:
Done. An `async_hook` is used for tracking async uncaught exceptions.